obs_maven/repo.py

```diff
@@ -59,8 +60,20 @@ def parse_primary(self):
         logging.debug("Parsing primary %s", primary_url)
         for cnt in range(1, 4):
             try:
-                with urllib.request.urlopen(primary_url) as primary_fd:
-                    with gzip.GzipFile(fileobj=primary_fd, mode="rb") as gzip_fd:
+                logging.debug("Getting primary try %s", cnt)
+
+                with tempfile.TemporaryFile() as tmp_file:
+                    # Read everything first to avoid connection resets
+                    with urllib.request.urlopen(primary_url) as primary_fd:
+                        # Avoid loading the complete document into memory
+                        chunk_size = 1000
```

Comment:
I have no idea what a good chunk size is for writing a HTTP body to a file. If anyone has resources or input, I'd be happy to learn how to optimize this. 

I think this should be fine to try it like it is.

Reply:
Ideally we should write the size we get from the request to avoid waiting for additional bites. Not sure how to handle that here though.... Unless this is the size of the chunks we get from the server. According to https://stackoverflow.com/questions/2811006/what-is-a-good-buffer-size-for-socket-programming/2811047#2811047 getting a proper size may be pretty hard.

I would leave it as is for now

Reply:
1kB seems way too small for me. How big are the failing packages? I think 1MB could be a good chunk.

Reply:
This part only downloads the primary.xml.gz, not the RPMs themselves. These files differ in size, based on the repository they are in. [This one for example](http://download.suse.de/ibs/SUSE:/SLE-15-SP2:/Update/standard/repodata/09e0229503839b93028f237950722aa11c4c5f5d907da0b0e82e8fc8461daa55-primary.xml.gz.mirrorlist) is on the larger side with 125MiB 

> I think 1MB could be a good chunk.

What do you base this number on? Not saying it's bad, I just want to learn how to come up with such numbers myself :innocent:

Reply:
For the record, MC suggested to start with 256KiB.

Reply:
Well, I'd put it in relation to the size of downloaded data as you want to have a low number of read requests.

Reply:
As I said, sizes vary a lot. Smaller repos have less entries in their primary.xml.gz. Looking at the Uyuni:Master repo, it's only 105KiB large.

Anyway, before we spend too much time on this point, I'm inclined to use 1MB which will cover the smaller files in one go and does not cause too many requests for larger ones. 

Thanks for the discussion, I learned something!